plugins/env/app/views/external_environment_variable_groups/_fields.html.erb

```diff
@@ -0,0 +1,34 @@
+<% external_env_groups =  form.object.external_environment_variable_groups %>
+<%= form.fields_for :external_environment_variable_groups, external_env_groups + [ExternalEnvironmentVariableGroup.new] do |fields| %>
+  <div class="form-group">
+    <div class="col-lg-2">
+      <%= fields.text_field :name, class: "form-control monospace", placeholder: "Name" %>
+    </div>
+
+    <div class="col-lg-3">
+      <%# using a text area so users can resize them with browser controls %>
+      <%= fields.text_area :description, class: "form-control monospace validate_whitespace", rows: 1, placeholder: "Description" %>
+    </div>
+
+    <div class="col-lg-5">
+      <%= fields.text_field :url, class: "form-control monospace", placeholder: ExternalEnvironmentVariableGroup::S3_URL_FORMAT %>
```

Comment:
FYI there is in-browser regex matching too, I thin `def input` in form-builder supports that